sdx-submitter.go

```diff
@@ -0,0 +1,168 @@
+package main
+import(
+		"fmt"
+		"flag"
+		"github.com/streadway/amqp"
+		"os"
+		"io/ioutil"
+)
+
+func main() {
+
+	// Read a source file and place it on a rabbit topic exchange
+	var url, exchange, routing_key,    encryption_key,    signing_key, name, password string
+	var port, host , queue, vhost, message_path_and_file string
+	var msg_body []byte
+	var err error
+
+	getArguments(&name, &password, &port, &host, &url, &queue, &exchange, &routing_key, &encryption_key, &signing_key, &vhost, &message_path_and_file)
+
+	msg_body = getBody(message_path_and_file)
+
+	// If encyrpt specified then encrypt
+	// if sign specified then sign
+
+	sendToRabbit(url, exchange, queue, routing_key, err, msg_body)
+	
+}
+
+func getArguments(name *string,
+	password *string,
+	port *string,
+	host *string,
+	url *string,
+	queue *string,
+	exchange *string,
+	routing_key *string,
+	encryption_key *string,
+	signing_key *string,
+	vhost *string,
+	message_path_and_file *string)  {
+	// define the command line parameters
+	flag.StringVar(name, "n", "", "name of the rabbit user")
+	flag.StringVar(password, "p", "", "password of the rabbit user")
+	flag.StringVar(port, "rport", "", "port used to connect to rabbit")
+	flag.StringVar(host, "rhost", "localhost", "hostname used to connect to rabbit")
+	flag.StringVar(vhost, "v", "", "vhostname used to connect to rabbit")
+	flag.StringVar(url, "u", "", "url connection string ")
+	flag.StringVar(queue, "q", "", "name of the rabbit queue")
+	flag.StringVar(exchange, "x", "message", "name of the rabbit exchange")
+	flag.StringVar(routing_key, "r", "sdx-made-up-routing-key", "rabbit routing key")
+	flag.StringVar(encryption_key, "e", "sdx-made-encryption_key", "path to a private key file used for encryption")
+	flag.StringVar(signing_key, "s", "sdx-made-signing_key", "path to a private key used for signing")
+	flag.StringVar(message_path_and_file, "f", "", "path to filename to send")
+	flag.Parse()
+	// Use values from environment variables if no value empty
+	getFromEnvIfEmpty(name, "RABBITMQ_DEFAULT_USER", "guest")
+	getFromEnvIfEmpty(password, "RABBITMQ_DEFAULT_PASS", "guest")
+	getFromEnvIfEmpty(port, "RABBITMQ_PORT", "5672")
+	getFromEnvIfEmpty(host, "RABBITMQ_HOST", "rabbit")
+	getFromEnvIfEmpty(queue, "RABBIT_SURVEY_QUEUE", "rabbit")
+	getFromEnvIfEmpty(exchange, "RABBITMQ_EXCHANGE", "rabbit")
+	getFromEnvIfEmpty(vhost, "RABBITMQ_DEFAULT_VHOST", "%2f")
+	if *url == "" {
+		*url = fmt.Sprintf("amqp://%s:%s@%s:%s/%s", *name, *password, *host, *port, *vhost)
+	} else {
+		fmt.Println("Url use overrides specific parameters") //A specific URL is set , so use it
+	}
+}
+
+func getFromEnvIfEmpty(target *string, key string, defaultValue string)  {
+	if *target == "" {
+		if value, present := os.LookupEnv(key); present {
+			*target = value
```

Comment:
Don't set by reference if possible - prefer to return the value instead (and an error if appropriate).

Reply:
done